### PR TITLE
dev-python/clade: add 3.3.7

### PR DESCRIPTION
**Pull request by evdenis**
Add clade util for building a compilation database.
https://github.com/17451k/clade

### CODE REVIEW

File: dev-python/cchardet/metadata.xml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+<maintainer type="project">
```
Comment (thesamesam) [inline dev-python/cchardet/metadata.xml:4]: We [shouldn't](https://wiki.gentoo.org/wiki/Project:Python#Adding_packages) add python@ to projects unless you've agreed it with them (or any project).

File: dev-python/cchardet/metadata.xml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (thesamesam) [inline dev-python/cchardet/metadata.xml:1]: Indent with tabs? Also, add proxy-maint.

File: dev-python/clade/clade-3.3.7.ebuild [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{7..9} )
+DISTUTILS_USE_SETUPTOOLS=rdepend
+
+inherit distutils-r1
+
+DESCRIPTION="Tool for extracting information about software build process and source code"
+HOMEPAGE="https://pypi.org/project/clade/"
```
Comment (thesamesam) [inline dev-python/clade/clade-3.3.7.ebuild:12]: Add GitHub to [metadata](https://devmanual.gentoo.org/ebuild-writing/misc-files/metadata/index.html) (<remote type="github-id">).

File: dev-python/clade/clade-3.3.7.ebuild [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+KEYWORDS="~amd64"
+
+RDEPEND="
+	dev-python/ujson[${PYTHON_USEDEP}]
+	dev-python/chardet[${PYTHON_USEDEP}]
+	dev-python/cchardet[${PYTHON_USEDEP}]
+	dev-python/graphviz[${PYTHON_USEDEP}]
+	dev-python/ply[${PYTHON_USEDEP}]
+"
+BDEPEND="dev-util/cmake"
```
Comment (thesamesam) [inline dev-python/clade/clade-3.3.7.ebuild:26]: Tests too, [please](https://dev.gentoo.org/~mgorny/python-guide/distutils.html#enabling-tests)! Also, if we're using cmake, do we need to be using the eclass to pick up the goodness required (it does some special stuff to adapt to Gentoo)?

File: dev-python/clade/metadata.xml [hunk 5]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+<maintainer type="project">
```
Comment (thesamesam) [inline dev-python/clade/metadata.xml:4]: See earlier comments re Python project.

**Review by thesamesam — CHANGES_REQUESTED** [review]

File: dev-python/clade/clade-3.3.7.ebuild (hunk 4 above)
Comment (evdenis) [reply dev-python/clade/clade-3.3.7.ebuild:26]: > Tests too, please!

Tests will not work. They require additional external components to download.

> Also, if we're using cmake, do we need to be using the eclass to pick up the goodness required

I think no. setup.py script calls cmake internally (https://github.com/17451k/clade/blob/master/setup.py#L61-L67) to build C-libs for intercepting.

Comment (thesamesam) [reply dev-python/clade/clade-3.3.7.ebuild:26]: In that case, we normally ask upstream to make downloading optional / allow local copies, and try to download manually in SRC_URI conditionally based on test?. But if it's too much work to do that, we just ask them to make it optional/allow local copies.

Comment (evdenis) [reply dev-python/clade/clade-3.3.7.ebuild:26]: I already asked the author to make tests that depends on external components conditional.

**Pull request closed without merging**